### PR TITLE
GroupStreak data model and repository layer

### LINKED ISSUE #384: Streak Data Model & Repository
Create `GroupStreak.kt` data class with the following fields:

- `groupId: String`: the group this streak belongs to
- `userId: String`: the user this streak belongs to
- `currentStreakWeeks: Int`: number of consecutive weeks in current streak
- `currentStreakActivities: Int`: total activities joined during current streak
- `currentStreakStartDate: Timestamp`: Monday of the first week of current streak
- `lastActiveWeekStart: Timestamp`: Monday of the most recent week with activity (used to detect broken streaks)
- `bestStreakWeeks: Int`: highest consecutive weeks ever achieved
- `bestStreakActivities: Int`: total activities joined during best streak

Create `GroupStreakRepository` interface with:

- `getStreaksForGroup(groupId: String): List<GroupStreak>` for leaderboard
- `getStreakForUser(groupId: String, userId: String): GroupStreak?`
- `updateStreak(groupId: String, userId: String, streak: GroupStreak)`

Create `GroupStreakRepositoryFirestore` implementation storing documents at `groups/{groupId}/streaks/{userId}` since each user has one GroupStreak per group.

Create `GroupStreakRepositoryLocal` in-memory implementation for testing and offline mode, using a map keyed by `{groupId}_{userId}`.

Create `GroupStreakRepositoryProvider` singleton for dependency injection, returning `LocalRepository` in test environments and `FirestoreRepository` in production.

### PR DESCRIPTION
**Pull request by BryanDMP**
# GroupStreak data model and repository layer

Implements the foundation for individual user streaks within groups, enabling future leaderboard display. (c.f. Figma)

---

### Changes
- Add `GroupStreak` data class with current/best streak tracking fields
- Add `GroupStreakRepository` interface defining streak operations
- Add `GroupStreakRepositoryFirestore` storing documents at `groups/{groupId}/streaks/{userId}`
- Add `GroupStreakRepositoryLocal` in-memory implementation for testing
- Add `GroupStreakRepositoryProvider` singleton for dependency injection

---

### Testing
- `GroupStreakTest`: 2 tests covering data class construction
- `GroupStreakRepositoryFirestoreTest`: 6 tests covering Firestore operations and edge cases
- `GroupStreakRepositoryLocalTest`: 8 tests covering in-memory operations
- `GroupStreakRepositoryProviderTest`: 6 tests covering environment detection and singleton behavior

---

### What's next
The next step is to implement `StreakService` which will handle the streak logic: determining week boundaries (Monday 00:00 UTC to Sunday 23:59 UTC), updating streaks when users join events or series, incrementing activity/week counters, detecting and resetting broken streaks, and updating best streak records when current exceeds best.

---

Closes #384

### CODE REVIEW

File: app/src/main/java/com/android/joinme/model/groups/streaks/GroupStreakRepositoryFirestore.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          groupId = groupId,
+          userId = doc.id,
+          currentStreakWeeks = doc.getLong("currentStreakWeeks")?.toInt() ?: 0,
+          currentStreakActivities = doc.getLong("currentStreakActivities")?.toInt() ?: 0,
+          currentStreakStartDate = doc.getTimestamp("currentStreakStartDate") ?: Timestamp.now(),
+          lastActiveWeekStart = doc.getTimestamp("lastActiveWeekStart") ?: Timestamp.now(),
+          bestStreakWeeks = doc.getLong("bestStreakWeeks")?.toInt() ?: 0,
+          bestStreakActivities = doc.getLong("bestStreakActivities")?.toInt() ?: 0)
+    } catch (_: Exception) {
+      null
```
Comment (AlexisPDS) [inline app/src/main/java/com/android/joinme/model/groups/streaks/GroupStreakRepositoryFirestore.kt]: i think it would be useful to add a Log statement for error handling.

**Review by AlexisPDS — APPROVED** [review]
Thanks for the quick and efficient changes.
This Pr looks good to me
I will merge it.
